### PR TITLE
Add docs for sharded descriptions

### PR DESCRIPTION
**Pull request by notriddle**
r? @GuillaumeGomez

### CODE REVIEW

File: src/rustdoc-internals/search.md [hunk 1]
```diff
@@ -13,29 +13,55 @@ scans them linearly to search.
 
 `search.js` calls this Raw, because it turns it into
 a more normal object tree after loading it.
-Naturally, it's also written without newlines or spaces.
+For space savings,
+it's also written without newlines or spaces.
```
Comment (GuillaumeGomez) [inline src/rustdoc-internals/search.md:17]: I think you can put it on the same line as the previous sentence. ;)

File: src/rustdoc-internals/search.md [hunk 2]
```diff
@@ -13,29 +13,55 @@ scans them linearly to search.
...
 
+| Key | Name                 | Description  |
+| --- | -------------------- | ------------ |
+| `n` | Names                | Item names   |
+| `t` | Item Type            | One-char item type code |
+| `q` | Parent module        | `Map<index, path>` |
+| `i` | Parent type          | list of indexes |
+| `f` | Function signature   | [encoded](#i-f-and-p) |
+| `b` | Impl disambiguator   | `Map<index, string>` |
+| `c` | Deprecation flag     | [roaring bitmap](#roaring-bitmaps)
```
Comment (GuillaumeGomez) [inline src/rustdoc-internals/search.md:59]: Can you keep the ending `|` for coherency please? Same for below lines.

File: src/rustdoc-internals/search.md [hunk 3]
```diff
@@ -78,36 +104,45 @@ It makes a lot of compromises:
...
+| 2 | `Data` | `F` | The data struct | `crate_name` | 0 | `` ` `` | NULL | 0 |
+
+[d]: #how-descriptions-are-stored
+
+The crate row is implied in most columns, since its type is known (it's a crate),
+it can't have a parent (crates form the root of the module tree),
+its name is specified as the map key,
+and function-specific data like the impl disambiguator can't apply either.
+However, it can still have a description and it can still be deprecated.
+The crate is, therefore, has a primary key of `0`.
```
Comment (GuillaumeGomez) [inline src/rustdoc-internals/search.md:127]: The `is` is too much here I think.

**Review by GuillaumeGomez — COMMENTED** [review]
Apart from the small typos, looks good to me. This update is very welcome!

**Review by fmease — APPROVED** [review]

**Pull request merged**